### PR TITLE
Add RestoreVolume spec for cloudbackup restore api

### PR DESCRIPTION
**Pull request by talakad**
Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:
Allows users to modify some of the volume spec related fields for the volume being restored from cloudbackup. (Current scheme does not support things like specifying HA level different than original volume) 
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -4438,3 +4442,44 @@ enum EnforcementType {
...
+  // Cos specifies the relative class of service.
+  CosType cos = 6;
+  // IoProfile provides a hint about application using this volume.
+  IoProfile io_profile = 7;
+  // SnapshotInterval in minutes, set to 0 to disable snapshots
+  uint32 snapshot_interval = 9;
+  // (deprecated, do not use) VolumeLabels configuration labels
+  map<string, string> volume_labels = 10;
+  // ReplicaSet is the desired set of nodes for the volume data.
+  ReplicaSet replica_set = 12;
```
Comment (disrani-px) [inline api/api.proto]: Why have gaps in the fields?

File: api/api.proto [hunk 2]
```diff
@@ -4438,3 +4442,44 @@ enum EnforcementType {
...
+message RestoreVolumeSpec {
+  // HaLevel specifies the number of copies of data.
+  int64 ha_level = 5;
+  // Cos specifies the relative class of service.
+  CosType cos = 6;
+  // IoProfile provides a hint about application using this volume.
+  IoProfile io_profile = 7;
+  // SnapshotInterval in minutes, set to 0 to disable snapshots
+  uint32 snapshot_interval = 9;
+  // (deprecated, do not use) VolumeLabels configuration labels
```
Comment (disrani-px) [inline api/api.proto]: Do we need this if it is deprecated?

File: api/api.proto [hunk 3]
```diff
@@ -4438,3 +4442,44 @@ enum EnforcementType {
...
+  map<string, string> volume_labels = 10;
+  // ReplicaSet is the desired set of nodes for the volume data.
+  ReplicaSet replica_set = 12;
+  // Aggregation level Specifies the number of parts the volume can be aggregated from.
+  uint32 aggregation_level = 13;
+  // SnapshotSchedule a well known string that specifies when snapshots should be taken.
+  string snapshot_schedule = 16;
+  // Scale allows autocreation of volumes.
+  uint32 scale = 17;
```
Comment (disrani-px) [inline api/api.proto]: Don't think we need this

**Comment by talakad** [comment]
Thanks @disrani-px . I realize i had the wrong branch for review. Closing this and will open a new one. That should address these comments.

**Pull request closed without merging**